backend/src/main/java/com/google/models/Anomaly.java

```diff
@@ -118,6 +133,11 @@ public Entity toEntity() {
         dataPointsEntity.setProperty(timestamp.toString(), (long) metricValue.getValue()));
     anomalyEntity.setProperty(DATA_POINTS_PROPERTY, dataPointsEntity);
 
+    List<EmbeddedEntity> relatedDataEntityList = relatedDataList.stream()
+        .map(relatedData -> relatedData.toEmbeddedEntity())
```

Comment:
RelatedData::toEmbeddedEntity?

https://docs.google.com/document/d/1M4eVysmUlz6caWTBtMIl_R8vWxWVQqXhgXswa5mxBgU/edit?hl=en for more infromation about method references for optional reading